src/main/java/uk/nhs/hee/tis/trainee/forms/model/AbstractAuditedForm.java

```diff
@@ -107,47 +108,41 @@ public void setLifecycleState(LifecycleState lifecycleState, Status.StatusDetail
         .revision(revision)
         .build();
 
+    Instant submitted;
+
+    if (lifecycleState == LifecycleState.SUBMITTED) {
+      submitted = statusInfo.timestamp;
+    } else {
+      submitted = status != null ? status.submitted : null;
```

Comment:
So we preserve a previous `submitted` timestamp value if a form is subsequently unsubmitted/rejected, and only update `submitted` when the form is resubmitted?

Reply:
Yes, it's meant mostly as an indication when a form became "ready" to reivew/approve I guess. The original submission timestamp isn't as useful as knowing when it was submitted "correctly"